cellengine/resources/compensation.py

```diff
@@ -1,20 +1,50 @@
 from __future__ import annotations
+from dataclasses import dataclass, field
+from typing import List, TYPE_CHECKING
+
+from dataclasses_json.cfg import config
 import numpy
 import pandas
-import typing
+from pandas.core.frame import DataFrame
 
 import cellengine as ce
-from cellengine.payloads.compensation import _Compensation
+from cellengine.utils.dataclass_mixin import DataClassMixin
+
 
-if typing.TYPE_CHECKING:
-    from cellengine.payload.fcs_file import FcsFile
+if TYPE_CHECKING:
+    from cellengine.resources.fcs_file import FcsFile
 
 
-class Compensation(_Compensation):
-    """A class representing a CellEngine compensation matrix. Can be applied to
-    FCS files to compensate them.
+@dataclass
+class Compensation(DataClassMixin):
+    """A class representing a CellEngine compensation matrix.
+
+    Can be applied to FCS files to compenlsate them.
     """
 
+    _id: str = field(metadata=config(field_name="_id"))
+    name: str
+    experiment_id: str
+    channels: List[str]
+    dataframe: DataFrame = field(
+        metadata=config(
+            field_name="spillMatrix",
+            encoder=lambda x: x.to_numpy().flatten().tolist(),
+            decoder=lambda x: DataFrame(data=numpy.array(x)),
+        )
+    )
+
+    def __post_init__(self):
+        self.dataframe = pandas.DataFrame(
+            data=numpy.array(self.dataframe).reshape(self.N, self.N),
+            columns=self.channels,
+            index=self.channels,
+        )
```

Comment:
Mhmm, I sure did, thanks for catching that unnecessary round-trip. I used `post_init` here because `self.N` is not available to the decoder.  I _could_ that I can use the sum of values in the spill matrix to get `N`, but I still need to assign the column and index names, which is slow.

Option 1:
```python
    dataframe: DataFrame = field(
        metadata=config(
            field_name="spillMatrix",
            encoder=lambda x: x.to_numpy().flatten().tolist(),
            decoder=lambda x: numpy.array(x),
        )
    )

    def __post_init__(self):
        self.dataframe = DataFrame(
            self.dataframe.reshape(self.N, self.N),
            columns=self.channels,
            index=self.channels,
        )
```

Option 2
```python
    dataframe: DataFrame = field(
        metadata=config(
            field_name="spillMatrix",
            encoder=lambda x: x.to_numpy().flatten().tolist(),
            decoder=lambda x: DataFrame(
                numpy.array(x).reshape(sum(x), sum(x)),
            )
        )
    )

    def __post_init__(self):
        self.dataframe.columns = self.channels
        self.dataframe.index = self.channels
```

Option 2 is ~0.3 ms slower on average